### PR TITLE
Momsfirstname

### PR DESCRIPTION
**Pull request by saulakravitz**
Added support for Mom's First Name (DMOMF)

### CODE REVIEW

**Comment by saulakravitz** [comment]
- update jurisdictionID extension URL
- BirthRecordIdentifier encoding in ISO 3166-2
- Several changes to the codesystem (from URL to PH OID) for Y/N codes
-

File: VRDR/IJEMortality.cs [hunk 1]
```diff
@@ -2491,21 +2492,57 @@ public string IDOB_YR
...
-                // NOOP
+                if (!String.IsNullOrWhiteSpace(value))
+                {
+                    String birthCountry = BPLACE_CNT;
+                    String ISO31662code;
+                    Console.Error.WriteLine("BSTATE.set with country = " + birthCountry);
+                    switch (value){
+                        case "ZZ": // UNKNOWN OR BLANK U.S. STATE OR TERRITORY OR UNKNOWN CANADIAN PROVINCE OR UNKNOWN/ UNCLASSIFIABLE COUNTRY
+                            // Console.Error.WriteLine("BSTATE.set with value ZZ");
+                            return;  // do nothing
```
Comment (rbclark) [inline VRDR/IJEMortality.cs:2529]: If this setter is called on a record which already has a state set will it clear the state or leave it set?

File: VRDR/IJEMortality.cs [hunk 2]
```diff
@@ -81,6 +81,7 @@ public IJEMortality(string ije, bool validate = true)
                 // Grab the field value
                 string field = ije.Substring(info.Location - 1, info.Length);
                 // Set the value on this IJEMortality (and the embedded record)
+                // Console.Error.WriteLine(" field= " + field + "offset = " + info.Location);
```
Comment (rbclark) [inline VRDR/IJEMortality.cs:84]: This PR seems to have a lot of commented out Console Writelines and other code, my suggestion would be remove all the commented out code.

File: VRDR/IJEMortality.cs [hunk 3]
```diff
@@ -2491,21 +2492,57 @@ public string IDOB_YR
...
+                            break;
+                        default:  // a 2 character state
+                            if (birthCountry.Equals("US") || birthCountry.Equals("CA")){
+                                // Console.Error.WriteLine("BSTATE.set with US or CA" );
+                                ISO31662code = birthCountry + "-" + value;
+                            }else{
+                                ISO31662code = value;
+                            }
+                            // Console.Error.WriteLine("BSTATE.set with value " + ISO31662code );
+                            break;
```
Comment (rbclark) [inline VRDR/IJEMortality.cs:2542]: This seems like it opens up for the possibility of the birthCountry getting out of sync with the BirthRecordState, if the country is updated is the BSTATE also fixed to have the proper birthCountry?

File: VRDR/IJEMortality.cs [hunk 4]
```diff
@@ -3266,7 +3304,7 @@ public string ZIP9_R
                 }
             }
         }
-
+
```
Comment (rbclark) [inline VRDR/IJEMortality.cs:3307]: You may want to enable 'Trim trailing whitespace on save' in VSCode.

File: VRDR.Tests/DeathRecord_Should.cs [hunk 5]
```diff
@@ -1505,13 +1497,12 @@ public void Get_BirthRecordId()
         public void Set_BirthRecordState()
         {
             Dictionary<string, string> brs = new Dictionary<string, string>();
-            brs.Add("code", "MA");
-            brs.Add("system", "urn:iso:std:iso:3166:-2");
-            brs.Add("display", "Massachusetts");
+            Dictionary<string, string> iladdress = new Dictionary<string, string>();
+            brs.Add("code", "US-MA");
```
Comment (rbclark) [inline VRDR.Tests/DeathRecord_Should.cs:1501]: Would having a test for just `MA` still be useful here as well just to test the additional codepaths that were added?

**Review by rbclark — COMMENTED** [review]
Could you link to the related JIRA tickets so it is more clear which ones specifically this intends to close?

File: VRDR/IJEMortality.cs [hunk 6]
```diff
@@ -3352,7 +3392,27 @@ public string DMIDDLE
...
+        /// <summary>Mother's First Name</summary>
+        [IJEField(168, 1958, 50, "Mother's First Name", "DMOMF", 1)]
+        public string DMOMF
+        {
+            get
+            {
+                string[] names = record.MotherGivenNames;
+                if (names != null && names.Length > 0)
+                {
+                    return names[0];
```
Comment (smacadam) [inline VRDR/IJEMortality.cs:3404]: Should this be left justified so it is the correct length?

File: VRDR/DeathRecord.cs [hunk 7]
```diff
@@ -7292,6 +7276,55 @@ private Address DictToAddress(Dictionary<string, string> dict)
             return address;
         }
 
+       /// <summary>Convert an "address" dictionary to a FHIR Address coded for a Decedent's Residence.</summary>
+        /// <param name="dict">represents an address.</param>
+        /// <returns>the corresponding FHIR Address representation of the address.</returns>
+        private Address DictToAddressDecedentAddress(Dictionary<string, string> dict)
```
Comment (smacadam) [inline VRDR/DeathRecord.cs:7282]: I don't see this used anywhere, is this going to be used in the future?

**Pull request merged**